### PR TITLE
Fixes to quotes and invoices

### PR DESCRIPTION
**Pull request by SteveIreland**
I stumbled across a few issues related to quote and invoice processing.  This pull request addresses them.  Primarily the subtotal wasn't being calculated properly.  I'm not sure when that broke but it should be fixed now (at least for a sales tax environment ... I haven't tested it yet in a VAT tax scenario but will do that soon.

Due to a client request for invoices (actually quotes) I added the ability to distinguish between a normal cash or credit payment and a cash or credit payment to be treated as a deposit.

I know there was some discussion about representing payments and balance due as negative amounts.  After reviewing "standard practices" I went ahead and changed them to be rendered as positive amounts.

### CODE REVIEW

File: application/views/sales/invoice.php [hunk 1]
```diff
@@ -183,7 +183,7 @@ function(response)
         <tr>
             <td colspan="3" class="blank"> </td>
             <td colspan="2" class="total-line"> <textarea rows="5" cols="6"><?php echo $this->lang->line($amount_change >= 0 ? ($only_sale_check ? 'sales_check_balance' : 'sales_change_due') : 'sales_amount_due') ; ?></textarea></td>
-            <td class="total-value"><textarea rows="5" cols="6" id="change"><?php echo to_currency($amount_change); ?></textarea></td>
+            <td class="total-value"><textarea rows="5" cols="6" id="change"><?php echo to_currency(abs($amount_change)); ?></textarea></td>
```
Comment (daN4cat) [inline application/views/sales/invoice.php]: I'll revert this change when I merge the code. I also need to fix a couple of things that are not standard / consistent with the rest of the code style.

File: application/views/sales/quote.php [hunk 2]
```diff
@@ -111,7 +111,7 @@ function(response)
 			?>
             <tr class="item-row">
                 <td><?php echo $item['item_number']; ?></td>
-                <td class="item-name"><textarea rows="4" cols="6"><?php echo ($item['is_serialized'] || $item['allow_alt_description']) && !empty($item['description']) ? $item['description'] : $item['name']; ?></textarea></td>
+                <td class="item-name"><textarea rows="4" cols="6"><?php echo $item['name']; ?></textarea></td>
```
Comment (daN4cat) [inline application/views/sales/quote.php]: ok, many changes that are not so generic. @SteveIreland it's worth reconsidering this change.

File: application/language/en/sales_lang.php [hunk 3]
```diff
@@ -19,6 +20,7 @@
 $lang["sales_confirm_cancel_sale"] = "Are you sure you want to clear this sale? All items will cleared.";
 $lang["sales_confirm_delete"] = "Are you sure you want to delete the selected sales?";
 $lang["sales_credit"] = "Credit Card";
+$lang["sales_credit_deposit"] = "Credit Deposit";
```
Comment (SteveIreland) [inline application/language/en/sales_lang.php]: I can see where this change might be particularly unique to my requirements so if you want please kill this pull request and I'll come back later with more generic changes and fixes.

**Comment by daN4cat** [comment]
@SteveIreland do you plan to submit in a different PR? I remember there were more fixes than your simple change. Not sure those should be submitted or not?

**Comment by SteveIreland** [comment]
Yes, I am reworking it.  The changes will provide better support for a services and repair company.  The feedback I've received is from someone who is building a startup using OSPOS.  The number of employees ranges from 5 to 9 so this should be a perfect fit for the software.  I'm making sure that it is fully configurable.

I believe my changes are complete.  I'm just running it through various test scenarios at this point.  I have to have it installed in production by Monday.

**Pull request closed without merging**